core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -247,6 +252,18 @@ public Map<String, SnapshotRef> refs() {
     return refs;
   }
 
+  @Override
+  public MetricsReporter metricsReporter() {
+    if (lazyMetricsReporter == null) {
+      synchronized (this) {
+        if (lazyMetricsReporter == null) {
+          lazyMetricsReporter = CatalogUtil.loadMetricsReporter(this.metricsReporterProperties);
```

Comment:
this is actually wrong and this should pass `properties` so that we have `CatalogUtil.loadMetricsReporter(this.properties)`

To finish the initialization of the metrics reporter after it was loaded, you'd have to do `lazyMetricsReporter.initialize(this.metricsReporterProperties)`

Reply:
metrics reporter is configured by catalog properties, not table properties. I add the interface method `MetricsReporter.properties()` to return the  properties which used to load and initialize this `MetricsReporter`,  therefor the `SerializableTable` can use this properties to load and initialize `MetricsReporter` agagin

Reply:
sorry my bad, you're right. it should be the catalog properties